openapi/generator.go

```diff
@@ -895,7 +895,17 @@ func (g *Generator) newSchemaFromType(t reflect.Type) *SchemaOrRef {
 		case reflect.Slice, reflect.Array, reflect.Map:
 			return g.buildSchemaRecursive(t)
 		case reflect.Struct:
-			return g.newSchemaFromStruct(t)
+			sor := g.newSchemaFromStruct(t)
```

Comment:
We already check if the type is "nullable" above when dereferencing the pointer. We could simply check if the schema pointed by the ref is non-nil and set the `Nullable` property accordingly.